upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -375,6 +375,40 @@ func (c *awsCloudImplementation) GetCloudGroups(cluster *kops.Cluster, instanceg
 	return getCloudGroups(c, cluster, instancegroups, warnUnmatched, nodes)
 }
 
+// GetCloudGroupMessages returns messages from a groups of autoscale groups
+// TODO better comments
+func (c *awsCloudImplementation) GetCloudGroupMessages(cluster *kops.Cluster, instancegroups []*kops.InstanceGroup, warnUnmatched bool, nodes []v1.Node) ([]*cloudinstances.CloudInstanceGroupMemberMessage, error) {
```

Comment:
How about we add `fetchMessages bool` to `GetCloudGroups(cluster *kops.Cluster, instancegroups []*kops.InstanceGroup, warnUnmatched bool, nodes []v1.Node)`, and run this through that function?  We can messages (or events) to `CloudInstanceGroup`

Reply:
Fair.  I need to figure out how to get it running before nodes exist though.  This should run when then API is not up yet.  So because of that, I may need another call.  The code still is not working yet for nodes, and I need to change GCE to not error out as well.